src/originalBudgetTrackingApp/transactions/transactions.ts

```diff
@@ -0,0 +1,114 @@
+import { uniq } from 'lodash';
+import { EnrichedTransaction } from '../commonTypes';
+import { Transaction } from '../import/bankScraper';
+import { compareObjects } from './dates';
+
+/*
+{
+  success: boolean,
+  accounts: [{
+    accountNumber: string,
+    txns: [{
+      type: string, // can be either 'normal' or 'installments'
+      identifier: int, // only if exists
+      date: string, // ISO date string
+      processedDate: string, // ISO date string
+      originalAmount: double,
+      originalCurrency: string,
+      chargedAmount: double,
+      description: string,
+      memo: string, // can be null or empty
+      installments: {
+        number: int, // the current installment number
+        total: int, // the total number of installments
+      },
+      status: string //can either be 'completed' or 'pending'
+    }],
+  }],
+  errorType: "invalidPassword"|"changePassword"|"timeout"|"generic", // only on success=false
+  errorMessage: string, // only on success=false
+}
+*/
+
+// export const properties = [
+//   {
+//     name: 'type',
+//     title: 'Type',
+//   },
+//   {
+//     name: 'identifier',
+//     title: 'Identifier',
+//   },
+//   {
+//     name: 'date',
+//     title: 'Date',
+//     column: true,
+//     hash: (value) => unixMilli(value),
+//   },
+//   {
+//     name: 'processedDate',
+//     title: 'Processed Date',
+//   },
+//   {
+//     name: 'originalAmount',
+//     title: 'Original Amount',
+//   },
+//   {
+//     name: 'originalCurrency',
+//     title: 'Original Currency',
+//   },
+//   {
+//     name: 'chargedAmount',
+//     title: 'Charged Amount',
+//     column: true,
+//     hash: true,
+//   },
+//   {
+//     name: 'description',
+//     title: 'Description',
+//     column: true,
+//     hash: (value) => calculateHash(value),
+//   },
+//   {
+//     name: 'memo',
+//     title: 'Memo',
+//   },
+//   {
+//     name: 'number',
+//     title: 'Installments Number',
+//   },
+//   {
+//     name: 'total',
+//     title: 'Installments Total',
+//   },
+//   {
+//     name: 'status',
+//     title: 'Status',
+//   },
+// ];
+
+const transactionArrayToObject = (transactions: EnrichedTransaction[]) => transactions.reduce((acc, enrichedTransaction) => {
+  acc[enrichedTransaction.hash] = enrichedTransaction;
+  return acc;
+}, {} as Record<string, EnrichedTransaction>);
+
+// TODO removed company and account, need attention
```

Comment:
See https://github.com/brafdlog/budget-tracking/pull/112/files#r501438587
Please do not resolve until I removing this `TODO`